WalletWasabi/WabiSabi/Client/AmountDecomposer.cs

```diff
@@ -65,12 +65,49 @@ public IEnumerable<Money> Decompose(IEnumerable<Coin> myInputCoins, IEnumerable<
 				}
 			}
 
-			if (remaining >= MinimumAmountPlusFee)
+			if (remaining >= MinimumAmountPlusFee && remainingVsize >= OutputSize)
```

Comment:
Isn't this second condition dangerous. I don't think it'd happen too many times, but if it does the user can lose considerable amount of money?

Reply:
It should be guaranteed that we can add one more Output by: 
https://github.com/zkSNACKs/WalletWasabi/pull/6908/files#diff-c61acf4e52eacd55e4c8309552f51ef996397593dd1b9935ad33f57c5abda1e7R51

IMO if the condition `remainingVsize >= OutputSize` should be moved into the if condition and throw an exception if not enough `remainingVsize`.

Reply:
https://github.com/zkSNACKs/WalletWasabi/pull/6908/commits/2527199db6ab58910a5f30b59e024d0ca68f28a7